### PR TITLE
Implement get_all function to return non-folding set-cookie headers

### PR DESCRIPTION
**Pull request by nakamura-shuta**
issue : https://github.com/cloudflare/workers-rs/issues/423

Implement get_all method for Headers using wasm-bindgen

This PR implements the get_all method for Headers with the following approach:

1. Define the get_all method in worker_sys::ext::headers::Headers using wasm-bindgen
2. Create a HeadersExt trait in worker_sys to extend web_sys::Headers functionality
3. Implement get_all in worker::Headers using the worker_sys implementation

Key changes:
- Add a new file worker-sys/src/ext/headers.rs to define Headers type and get_all method
- Create HeadersExt trait to extend web_sys::Headers with get_all method
- Update worker/src/headers.rs to use the new HeadersExt trait
- Modify the get_all implementation in worker::Headers to use the new wasm-bindgen based method

Specification of the get_all method:
- Retrieves all values of a given header name
- Returns the result as Result<Vec<String>>
- Uses wasm-bindgen to directly call the JavaScript Headers.getAll() method

This implementation allows for retrieving multiple values for any header, not just 'Set-Cookie', 
providing more flexibility and aligning with the Web API's Headers.getAll() method.

### CODE REVIEW

**Comment by kflansburg** [comment]
Why limit only to `set-cookie` header?

**Comment by nakamura-shuta** [comment]
The workerd code I used as reference for implementation made an error when a header other than `set-cookie` was specified, so I implemented it as is.

https://github.com/cloudflare/workerd/blob/a265d5c22bf2a6bc6394aa54db517ff86d6c788c/src/workerd/api/http.c%2B%2B#L229-L242

**Comment by kflansburg** [comment]
I see, I can ask internally about this. 

If this is already implemented in the runtime, would it make sense to just add the `getAll` method to the wasm_bindgen Header extension type here https://github.com/cloudflare/workers-rs/blob/main/worker-sys/src/ext/headers.rs?

**Comment by nakamura-shuta** [comment]
Yes, that seems to be the simplest way.
I will try to implement it that way.

**Comment by kflansburg** [comment]
Can we invoke the JavaScript `getAll` method via `wasm-bindgen`, rather than re-implementing it in Rust?

File: worker-sys/src/ext/headers.rs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
```
Comment (kflansburg) [inline worker-sys/src/ext/headers.rs:3]: Does this need to be buried in a `glue` module like some of the other extensions?

**Review by kflansburg — COMMENTED** [review]
This looks great! Were you able to test this?

File: worker-sys/src/ext/headers.rs (hunk 1 above)
Comment (nakamura-shuta) [reply worker-sys/src/ext/headers.rs]: Yes, I have included it in the glue module to match the others. Included in the glue module along with others

**Comment by nakamura-shuta** [comment]
@kflansburg

> This looks great! Were you able to test this?

I created the following application under the example directory and verified its operation with curl.

```
/// create sample application. examples/cookie/src/lib.rs
use worker::*;

// Main entry point for the worker
#[event(fetch)]
pub async fn main(req: Request, env: Env, _ctx: worker::Context) -> Result<Response> {
    // Set up a router and define a route for "/test_cookies"
    Router::with_data(())
        .get_async("/test_cookies", test_cookies)
        .run(req, env)
        .await
}

// Handler function for the "/test_cookies" route
async fn test_cookies(req: Request, _ctx: RouteContext<()>) -> Result<Response> {
    // Retrieve all Set-Cookie headers from the input request
    let input_cookies = req.headers().get_all("Set-Cookie")?;

    // Create a new response
    let mut res = Response::ok("Cookies processed")?;

    // Add the input Set-Cookie headers to the new response
    for cookie in &input_cookies {
        // Append each Set-Cookie header to the response
        res.headers_mut().append("Set-Cookie", cookie)?;
    }

    // Log the request and response headers for debugging
    console_log!("Request Set-Cookie headers: {:?}", input_cookies);
    console_log!("Response headers: {:?}", res.headers());

    // Return the response
    Ok(res)
}
```

```
# curl command in console

% curl 'http://localhost:8787/test_cookies' \
  -H 'Set-Cookie: a=1; Expires=Wed, 21 Oct 2015 07:28:00 GMT' \
  -H 'Set-Cookie: b=2; Expires=Wed, 21 Oct 2015 07:28:00 GMT' \
  -v
* Host localhost:8787 was resolved.
> GET /test_cookies HTTP/1.1
> ・・・・・・・
> Set-Cookie: a=1; Expires=Wed, 21 Oct 2015 07:28:00 GMT
> Set-Cookie: b=2; Expires=Wed, 21 Oct 2015 07:28:00 GMT
>
< HTTP/1.1 200 OK
< Content-Length: 17
< Content-Type: text/plain; charset=utf-8
< Set-Cookie: a=1; Expires=Wed, 21 Oct 2015 07:28:00 GMT
< Set-Cookie: b=2; Expires=Wed, 21 Oct 2015 07:28:00 GMT
<
* Connection #0 to host localhost left intact
Cookies processed
```

```
# wrangler dev console

% cd workers-rs/examples/cookie
% npx wrangler dev

[wrangler:inf] Ready on http://localhost:8787
Request Set-Cookie headers: ["a=1; Expires=Wed, 21 Oct 2015 07:28:00 GMT", "b=2; Expires=Wed, 21 Oct 2015 07:28:00 GMT"]
Response headers: Headers {
content-type = text/plain; charset=utf-8
set-cookie = a=1; Expires=Wed, 21 Oct 2015 07:28:00 GMT
set-cookie = b=2; Expires=Wed, 21 Oct 2015 07:28:00 GMT
}
[wrangler:inf] GET /test_cookies 200 OK (12ms)
```

**Pull request merged**